doc/user/mgmtd.rst

```diff
@@ -0,0 +1,376 @@
+with the new FRR Centralised Management Framework.
+
+For organizing the management data to be owned by the FRR Management plane,
+management data is stored in YANG in compliance with a pre-defined set
+of YANG based schema. Data shall also be stored/retrieved in YANG format only.
+
+The MGMTd also acts as a separate computational entity for offloading much
+of the management related computational overload involved in maintaining of
+management data and processing of management requests, from individual
+component daemons (which can otherwise be a signficant burden on the individual
+components, affecting performance of its other functionalities).
+
+Lastly, the MGMTd works in-tandem with one (or more) MGMT Frontend
+Clients and a bunch of MGMT Backend Clients to realize the entirety
+of the FRR Management plane. Some of the advanatages of this new framework
+are:
+
+ 1. Consolidation and management of all Management data by a single entity.
+ 2. Better control over configuration validation, commit and rollback.
+ 3. Faster collection of configuration data (without needing to involve
+    individual component daemons).
+ 4. Offload computational burden of YANG data parsing and validations
+    of new configuration data being provisoned away from individual
+    component daemons
+ 5. Improve performance of individual component daemons while loading
+    huge configuration or retrieving huge operational dataset.
+
+The new FRR Management Daemon consists of the following sub-components:
+ - MGMT Frontend Interface
+ - MGMT Backend Interface
+ - MGMT Transaction Engine
+
+.. _mgmt_frntnd:
+
+MGMT Frontend Interface
+=======================
+
+The MGMT Frontend Interface is a bunch of message-based APIs that lets
+any UI/Frontend client to interact with the MGMT daemon to requests a
+set of management operations on a specific datastore/database.
+Following is a list of databases/datastores supported by the MGMT
+Frontend Interface and MGMTd:
+
+ - Candidate Database:
+
+  - Consists of configuration data items only.
+  - Data can be edited anytime using SET_CONFIG API.
+  - Data can be retrieved anytime using GET_CONFIG/GET_DATA API.
+
+ - Running Database:
+
+  - Consists of configuration data items only.
+  - Data cannot be edited using SET_CONFIG API.
+  - Data can only be modified using COMMIT_CONFIG API after which un-committed
+    data from Candidate database will be first validated and applied to
+    individualBackend component(s). Only on successful validation and apply on
+    all individual components will the new data be copied over to the Running
+    database.
+  - Data can be retrieved anytime using GET_CONFIG/GET_DATA API.
+
+ - Startup Database:
+
+  - Consists of configuration data items only.
+  - This is a copy of Running database that is stored in persistent
+    storage and is used to load configurations on Running database during
+    MGMT daemon startup.
+  - Data cannot be edited/retrieved directly via Frontend interface.
```

Comment:
There needs to be a little more documentation here, mostly regarding `frr_startup.json`.  (It's mentioned further down, but not sufficient.)

Could you please explain:
- when exactly is `frr_startup.json` written. Is it only `mgmt commit-apply`?
- when is it read? Only at startup?
- is it necessary for the user to back up this file? What happens when the file is lost or corrupted?
- what happens when the user deletes all config in `/etc/frr` but leaves/forgets `frr_startup.json` - will config come back even though it was removed from `/etc/frr`? Can there be a conflict?
- how does `frr version X.Y` & `frr profile <datacenter|traditional>` work?

These docs should be added ASAP, ideally before merge, if not then shortly after.